draft-ietf-ace-oscore-profile.xml

```diff
@@ -546,7 +612,7 @@ OSCORE_Security_Context = {
       <section anchor="c-rs" title="C-to-RS: POST to authz-info endpoint">
 
         <t>
-          The client MUST use CoAP and the Authorization Information resource as described in section 5.8.1 of <xref target="I-D.ietf-ace-oauth-authz"/> to transport the token to the RS.
+          The client MUST generate a nonce N1 with a good amount of randomness. This profile RECOMMENDS to use a nonce of 64 bits. The client MUST store this nonce as long as the response from the RS is not received and the access token related to it is still valid. The client MUST use CoAP and the Authorization Information resource as described in section 5.8.1 of <xref target="I-D.ietf-ace-oauth-authz"/> to transport the token and N1 to the RS. The client MUST use the Content-Format "application/ace+cbor" defined in section 8.14 of <xref target="I-D.ietf-ace-oauth-authz"/>. The client MUST include the access token using the correct CBOR label (e.g., "cwt" for CWT, "jwt" for JWT) and N1 using the 'nonce' parameter defined in section 5.1.2 of <xref target="I-D.ietf-ace-oauth-authz"/>.
         </t>
```

Comment:
I have a problem with the term "good amount of randomness".   First I don't know how one would measure this to say it is satisfied.  Second, a client which does not have a good random number source, which is one of the things people keep trying to talk about, there is no way to satisfy this.

Reply:
Yes, the fact that we use nonces in this profile makes it so that we require good source of randomness. This is the same terminology used in https://core-wg.github.io/oscoap/draft-ietf-core-object-security.html#rfc.section.7.5 . If they don't have it, then something like EDHOC should be used, which was in this document before but got removed. How to phrase this so that it is acceptable?

Reply:
I don't think that going to EDHOC is going to make the nonces used any better.  What needs to be generated is a values which is 1) unlikely to have been previously used with this context and 2) is not highly predictable (i.e. not a counter).  I think that calling this entropy is a really incorrect notion.  The only important part is non-reuse.  The reason for saying random is just to "ensure" that this is going to be true.

Reply:
Ok, makes sense. I will try to rephrase using your 2 points. Although using terms such as "unlikely" and "not highly" can also incur in the same type of comments as "good amount". But yes what needs to be stressed is the non-reuse.

Reply:
Done in 9ffd22e.

Reply:
Another question: why the need for point 2, if we require that it is very unlikely to have been previously used with this context (also in cases of errors and reboot)? Where does the predictability matter here?

Reply:
I removed the part about predictability, because I do not see why that should be a requirement.